# **Tarea 02 - Calculo de medidas de tendencia central**

Jhonatan Yael Martínez Vargas - A01734193

Luis Adrián Díaz Santana González - A01734193

Francisco Antonio López Ricardez -A01737275

En la actualidad, el análisis de datos se está convirtiendo en una actividad recurrente dentro del mundo empresarial, pues permite examinar, transformar y modelar datos con el fin de extraer información útil. Esto hace que sea una herramienta muy útil al momento de optimizar procesos, tomar decisiones y descubrir patrones que puedan generar predicciones.

Por ese mismo motivo a continuación se presenta un breve ejemplo de cómo usando la librería pandas de python, se pueden extraer las medidas de tendencia central de una base de datos libre proveniente del e-commerce “Ulabox”, como estas mismas ayudan a entender cual es comportamiento del negocio.

---

### **Paso 01**

Se importa la libreria, se guardan los datos del archivo .csv y se muestra el dataset generado

In [24]:
import pandas as pd 

df = pd.read_csv("ulabox_orders_with_categories_partials_2017.csv"); df

customer  order  total_items  discount%  weekday  hour  Food%  Fresh%  \
0             0      0           45      23.03        4    13   9.46   87.06   
1             0      1           38       1.22        5    13  15.87   75.80   
2             0      2           51      18.08        4    13  16.88   56.75   
3             1      3           57      16.51        1    12  28.81   35.99   
4             1      4           53      18.31        2    11  24.13   60.38   
...         ...    ...          ...        ...      ...   ...    ...     ...   
29995     10235  29995            4       0.00        5    10   5.80    0.00   
29996     10236  29996            1       0.00        7    21   0.00    0.00   
29997     10237  29997           54       0.00        4    17   9.25    0.00   
29998     10237  29998           10       0.00        5    10   0.00    0.00   
29999     10238  29999            2       0.00        4    17   0.00    0.00   

       Drinks%  Home%  Beauty%  Health%  Baby%  Pets%  
0         3.48   0.00     0.00     0.00    0.0    0.0  
1         6.22   2.12     0.00     0.00    0.0    0.0  
2         3.37  16.48     6.53     0.00    0.0    0.0  
3        11.78   4.62     2.87    15.92    0.0    0.0  
4         7.78   7.72     0.00     0.00    0.0    0.0  
...        ...    ...      ...      ...    ...    ...  
29995    51.30   0.00     0.00     0.00    0.0   42.9  
29996     0.00   0.00   100.00     0.00    0.0    0.0  
29997    77.48  13.27     0.00     0.00    0.0    0.0  
29998   100.00   0.00     0.00     0.00    0.0    0.0  
29999     0.00   0.00     0.00     0.00  100.0    0.0  

[30000 rows x 14 columns]

Observando el data frame anterior nos percatamos que la base de datos con la que estamos trabajando contiene la informacion de **29999** compras realizadas, en este caso cada una de las columnas representa lo sigueinte:

- **Customer:** Numero de cliente.

- **Order:** Numero de pedido.

- **Total_items:** Total de objetos comprados.

- **Discont:** Porcentaje de descuento aplicado a la compra.

- **Weekdday:** Dia de la semana en la que se realizo la compra.

- **Hour:** Hora en la que se realizo la compra.

- **Food:** Porcentaje de alimentos que se compraron.

- **Fresh:** Porcentaje de productos frescos que se compraron.

- **Drinks:** Porcente de bebidas que se compraron.

- **Home:** Porcentaje de objetos para el hogar que se compraron.

- **Beauty:** Porcentaje de productos de belleza que se compraron.

- **Health:** Porcentaje de productos de cuidado de la salud que se compraron.

- **Baby:** Porcentaje de productos para bebe que se compraron.

- **Pets:** Porcentajes de productos para mascota que se compraron.


---

### **Paso 02**



In [25]:
df.describe()

customer         order   total_items     discount%       weekday  \
count  30000.000000  30000.000000  30000.000000  30000.000000  30000.000000   
mean    5012.483367  14999.500000     31.739933      8.455495      3.642367   
std     2888.646245   8660.398374     20.576579     14.199350      2.122031   
min        0.000000      0.000000      1.000000    -65.150000      1.000000   
25%     2541.750000   7499.750000     18.000000      1.177500      2.000000   
50%     5043.000000  14999.500000     29.000000      4.160000      3.000000   
75%     7483.000000  22499.250000     41.000000      9.990000      5.000000   
max    10238.000000  29999.000000    298.000000    100.000000      7.000000   

               hour         Food%        Fresh%       Drinks%         Home%  \
count  30000.000000  30000.000000  30000.000000  30000.000000  30000.000000   
mean      15.034667     23.560406     20.227279     23.349235     13.539575   
std        5.717052     21.719824     23.661767     22.523335     17.850282   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%       11.000000      7.220000      0.000000      6.710000      0.000000   
50%       15.000000     19.650000     10.930000     17.890000      7.210000   
75%       20.000000     33.260000     35.790000     33.162500     20.002500   
max       23.000000    100.000000    100.000000    100.000000    100.000000   

            Beauty%       Health%         Baby%         Pets%  
count  30000.000000  30000.000000  30000.000000  30000.000000  
mean       5.892949      1.132959     11.096145      1.043087  
std       13.996518      5.307620     24.740364      6.195390  
min        0.000000      0.000000      0.000000      0.000000  
25%        0.000000      0.000000      0.000000      0.000000  
50%        0.000000      0.000000      0.000000      0.000000  
75%        6.512500      0.000000      4.172500      0.000000  
max      100.000000    100.000000    100.000000    100.000000

In [26]:
df.mode()

customer  order  total_items  discount%  weekday  hour  Food%  Fresh%  \
0        6560.0      0         25.0        0.0      1.0  22.0    0.0     0.0   
1           NaN      1          NaN        NaN      NaN   NaN    NaN     NaN   
2           NaN      2          NaN        NaN      NaN   NaN    NaN     NaN   
3           NaN      3          NaN        NaN      NaN   NaN    NaN     NaN   
4           NaN      4          NaN        NaN      NaN   NaN    NaN     NaN   
...         ...    ...          ...        ...      ...   ...    ...     ...   
29995       NaN  29995          NaN        NaN      NaN   NaN    NaN     NaN   
29996       NaN  29996          NaN        NaN      NaN   NaN    NaN     NaN   
29997       NaN  29997          NaN        NaN      NaN   NaN    NaN     NaN   
29998       NaN  29998          NaN        NaN      NaN   NaN    NaN     NaN   
29999       NaN  29999          NaN        NaN      NaN   NaN    NaN     NaN   

       Drinks%  Home%  Beauty%  Health%  Baby%  Pets%  
0          0.0    0.0      0.0      0.0    0.0    0.0  
1          NaN    NaN      NaN      NaN    NaN    NaN  
2          NaN    NaN      NaN      NaN    NaN    NaN  
3          NaN    NaN      NaN      NaN    NaN    NaN  
4          NaN    NaN      NaN      NaN    NaN    NaN  
...        ...    ...      ...      ...    ...    ...  
29995      NaN    NaN      NaN      NaN    NaN    NaN  
29996      NaN    NaN      NaN      NaN    NaN    NaN  
29997      NaN    NaN      NaN      NaN    NaN    NaN  
29998      NaN    NaN      NaN      NaN    NaN    NaN  
29999      NaN    NaN      NaN      NaN    NaN    NaN  

[30000 rows x 14 columns]